### PR TITLE
Octopus deploy trance always marked failures

### PR DESCRIPTION
**Pull request by ids-pfinn**
### Description

Two changes:
1. A trace in the Octo.fs doesn't mark itself as successful before going out of scope so the output is always the `Failed` status
      - Prefix: image
      - Postfix: image

2. Adds additional functions to return the exit code of the octopus command.

## TODO

- [ ] New (API-)documentation for new features exist (Note: API-docs are enough, additional docs are in `help/markdown`)
- [ ] unit or integration test exists (or short reasoning why it doesn't make sense)
- [x] Change Process.execSimple to CreateProcess APIs - maybe.

### CODE REVIEW

**Comment by ids-pfinn** [comment]
@matthid The FAKE-CI failed due to the `TemplateIntegrationTests` step.
I went through the build output and I think the tests are passing
```
Passed: [36m 40
	Fake.Core.ProcessIntegrationTests/ProcessUtils.findLocalTool doesn't fail on non-existent path, #2390
	Fake.Core.ProcessIntegrationTests/Make sure cmd-echo works with regular redirect/lazy redirect
	Fake.Core.ProcessIntegrationTests/Make sure cmd-echo works with regular redirect/regular redirect
	Fake.Core.ProcessIntegrationTests/Make sure process with lots of output and error doesn't hang - #2401/lazy redirect
	Fake.Core.ProcessIntegrationTests/Make sure process with lots of output and error doesn't hang - #2401/regular redirect
	Fake.DotNet.MSBuild.IntegrationTests/#2112 (2)
	Fake.DotNet.MSBuild.IntegrationTests/#2112
	Fake.DotNet.MSBuild.IntegrationTests/#2134 (3)
	Fake.DotNet.MSBuild.IntegrationTests/#2134 (2)
	Fake.DotNet.MSBuild.IntegrationTests/#2134
	Fake.DotNet.NuGetIntegrationTests/getLastNuGetVersion works with myget - #2124
	Fake.DotNet.NuGetIntegrationTests/getLastNuGetVersion works - #2124
	Fake.DotNet.NuGetIntegrationTests/getLastNuGetVersion works - #2294
	Fake.DotNet.PaketIntegrationTests/findPaketExecutable works for global tools - #2361
	Fake.IO.FileIntegrationTests/Shell.mv works as expected when src is a dir dst is a dir - #2293
	Fake.IO.FileIntegrationTests/Shell.mv throws as expected when src is a dir and dst is a file - #2293
	Fake.IO.FileIntegrationTests/Shell.mv works as expected when src is a dir and dst doesnt exist - #2293
	Fake.IO.FileIntegrationTests/Shell.mv works as expected when src is a file dst is a dir - #2293
	Fake.IO.FileIntegrationTests/Shell.mv works as expected when src is a file and dst is a file - #2293
	Fake.IO.FileIntegrationTests/Shell.mv works as expected when src is a file and dst doesnt exist - #2293
	Fake.IO.FileIntegrationTests/Files created using File.create can be used immediately - #2183
	Fake.IO.FileIntegrationTests/File.tryGetVersion works when component is missing #2378
	Fake.IO.FileIntegrationTests/File.getVersion throws InvalidOperationException #2378
	Fake.Core.Globbing.IntegrationTests/glob should handle substring directories properly
	Fake.Core.Globbing.Tools.Tests/Test cannot find tool folder in sub path returns default
	Fake.Core.Globbing.Tools.Tests/Test find tool folder in sub path
	Fake.Core.Globbing.Tools.Tests/Test cannot find tool folder in sub path
	Fake.Core.Globbing.Tools.Tests/Test try find tool folder in sub path
	Fake.DotNet.CliIntegrationTests/Make sure dotnet installer works without spaces - #2319
	Fake.DotNet.CliIntegrationTests/Make sure dotnet installer can install into path with spaces - #2319
	Fake.DotNet.CliIntegrationTests/Make sure dotnet installer works in paths with spaces - #2319
	Fake.Core.IntegrationTests/issue #2007 - native libs work
	Fake.Core.IntegrationTests/issue #2025
	Fake.Core.IntegrationTests/reference fake core targets
	Fake.Core.IntegrationTests/use external paket.dependencies
	Fake.Core.IntegrationTests/context exists
	Fake.Core.IntegrationTests/reference fake runtime
	Fake.Core.IntegrationTests/simple runtime error
	Fake.Core.IntegrationTests/simple failed to compile
	Fake.Core.IntegrationTests/no dependencies hello world and casing #2314[37m
```
 
I ran the `TemplateIntegrationTests` step locally and the tests passed. 

Also I started down the path of switching the Process.execSimple call to the new CreateProcess APIs. But I was going to be modifying how the module processed the Command object and that seemed kinda big. Wanted to make sure I wasn't stepping on any toes before going forward with that.

**Comment by ids-pfinn** [comment]
Nvm about the previous comment. Looks like they are passing now.

**Comment by matthid** [comment]
Maybe a flaky test, the error was:

```
25h[21:52:53 ERR] Fake.DotNet.Cli.IntegrationTests.Template tests/can install and run the template/fails to build a target that doesn't exist failed in 00:01:09.5680000. 
The script should recognize the target doesn't exist. Actual value was false but had expected it to be true.
   at Fake.DotNet.Cli.IntegrationTests.TemplateTests.tests@153-3.Invoke(Unit unitVar) in D:\a\1\s\src\test\Fake.DotNet.Cli.IntegrationTests\TemplateTests.fs:line 157
   at Expecto.Impl.execTestAsync@960-1.Invoke(Unit unitVar) in /Users/h/dev/haf/expecto/Expecto/Expecto.fs:line 964
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 398
   at <StartupCode$FSharp-Core>.$Async.StartChild@1650-5.Invoke(AsyncActivation`1 ctxt) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 1650
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in E:\A\_work\130\s\src\fsharp\FSharp.Core\async.fs:line 109 <Expecto>

Failed:  1
	Fake.DotNet.Cli.IntegrationTests.Template tests/can install and run the template/fails to build a target that doesn't exist

```

Due to parallel build the output is further above. In any case I don't see this related to your changes. The Cli IntegrationTests are kind of flaky by design which is something I'm repeatedly think about but never come up with a good solution.

File: src/app/Fake.Tools.Octo/Octo.fs [hunk 1]
```diff
@@ -214,96 +214,119 @@ let private deployCommandLine (opts:DeployReleaseOptions) =
...
+    | 0 ->
+        __.MarkSuccess()
+        result
+    | _ ->
+        __.MarkFailed()
+        failwithf "Octo %s failed. Process finished with exit code %i" commandString result
+        result
+
+/// Creates a release and returns the exit code.
+let createReleaseWithExitCode setParams =
```
Comment (matthid) [inline src/app/Fake.Tools.Octo/Octo.fs]: Are you sure this is doing what you intended? From a quick look it looks like this either returns `0` or throws an exception (without actually testing it)

Comment (ids-pfinn) [reply src/app/Fake.Tools.Octo/Octo.fs]: I hadn't tested that yet but looking at it again after the weekend, yes this seems wrong.

**Comment by ids-pfinn** [comment]
@matthid I added 2 tests but wanted to make sure me making the internals visible to the test project was an okay pattern for you before adding the rest.

Also I wasn't sure the best way to handle the exception being throwing because it has the commandString in the message. I ended up creating the commandline again but I feel like a better solution exists.

**Comment by matthid** [comment]
> I added 2 tests but wanted to make sure me making the internals visible to the test project was an okay pattern for you before adding the rest.

Yes it is. Thanks a lot! Looks very good now. Let's see how it works out!

**Pull request merged**